### PR TITLE
Install kirk from repositories

### PR DESCRIPTION
**Pull request by acerv**
Now it's possible to define KIRK_FROM variable that can select kirk install from LTP repository or git.
This modification add support for transactional-update command.

- [osd verification run](https://openqa.suse.de/tests/11768749#step/test_memcontrol/1)

### CODE REVIEW

File: tests/kernel/kirk.pm [hunk 1]
```diff
@@ -30,11 +36,31 @@ sub run
...
+    if ($install_from =~ /git/i) {
+        if (is_transactional) {
+            trup_install("git");
+        } else {
+            zypper_call("in -y git");
+        }
+
+        assert_script_run("git clone -q --single-branch -b $branch --depth 1 $repo");
+
+        $cmd .= "python3 kirk/kirk ";
```
Comment (czerw) [inline tests/kernel/kirk.pm:51]: why `.=` and not just `=`?

File: tests/kernel/kirk.pm [hunk 2]
```diff
@@ -30,11 +36,31 @@ sub run
...
+    } else {
+        add_ltp_repo;
```
Comment (czerw) [inline tests/kernel/kirk.pm:53]: so we assume that kirk is in same repos as ltp, fair enough, but it should be mentioned is some doc section

File: tests/kernel/kirk.pm [hunk 3]
```diff
@@ -30,11 +36,31 @@ sub run
...
     select_serial_terminal;
+    zypper_call("ref");
+
+    my $cmd = '';
+    if ($install_from =~ /git/i) {
+        if (is_transactional) {
+            trup_install("git");
+        } else {
+            zypper_call("in -y git");
+        }
```
Comment (czerw) [inline tests/kernel/kirk.pm:47]: I would do one liner `is_transactional ? trup_install("git") : zypper_call("in -y git");`, but is just different style.

File: tests/kernel/kirk.pm (hunk 2 above)
Comment (acerv) [reply tests/kernel/kirk.pm:53]: You mean doc section inside the code or somewhere else?

File: tests/kernel/kirk.pm [hunk 4]
```diff
@@ -6,19 +6,28 @@
...
 use utils;
 use strict;
 use testapi;
 use warnings;
 use serial_terminal 'select_serial_terminal';
+use version_utils qw(is_transactional);
+use transactional 'trup_install';
+
+use FindBin 1.51 qw( $RealBin );
+use lib $RealBin;
```
Comment (czerw) [inline tests/kernel/kirk.pm:22]: why is needed findbin?

File: lib/LTP/utils.pm [hunk 5]
```diff
@@ -352,4 +353,67 @@ sub parse_runfiles {
...
+
+    my $run_cmd = is_transactional ? 'transactional-update -c -d --quiet run' : '';
+    for my $pkg (@pkgs) {
+        my $want_32bit = want_ltp_32bit($pkg);
+
+        record_info("LTP pkg: $pkg", script_output("$run_cmd rpm -qi $pkg | tee "
+                  . get_ltp_version_file($want_32bit)));
+        assert_script_run "find " . get_ltproot($want_32bit) .
+          q(/testcases/bin/openposix/conformance/interfaces/ -name '*.run-test' > )
+          . get_ltp_openposix_test_list_file($want_32bit);
```
Comment (czerw) [inline lib/LTP/utils.pm:415]: `get_ltp_openposix_test_list_file` will have to be moved as well

Comment (acerv) [reply lib/LTP/utils.pm:415]: `get_ltp_openposix_test_list_file` is already in LTP/utils.pm

**Pull request closed without merging**